pyomo/core/plugins/transform/interval_tightener.py

```diff
@@ -0,0 +1,67 @@
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.repn.canonical_repn import generate_canonical_repn
+from pyomo.util.plugin import alias
+import logging
+import textwrap
+logger=logging.getLogger('pyomo.core')
+class BoundsTightener(IsomorphicTransformation):
+    """Tightens bound on constraint, by iterating through each variable and
+    tightening the constraint bounds accordingly"""
+
+
+    alias('core.interval_tightener',doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+    def __init__(self):
+        super(BoundsTightener, self).__init__()
+
+    def _apply_to(self,instance):
+
+        for constr in instance.component_data_objects(ctype = Constraint,
+                                                        active = True,
+                                                        descend_into=True):
+            if not constr.body.polynomial_degree() == 1:
+                continue
+
```

Comment:
Looking at the code above, it seems that this transformation only applies to linear constraints. This caveat should appear in the code documentation for the class.